apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisChart.vue

```diff
@@ -53,38 +53,34 @@ const state = reactive({
 });
 
 /* api */
-let listCostAnalysisRequest: CancelTokenSource | undefined;
+const fetchCostAnalyze = getCancellableFetcher<CostAnalyzeDataModel>(SpaceConnector.client.costAnalysis.cost.analyze);
 const costQueryHelper = new QueryHelper();
 const listCostAnalysisData = async () => {
-    if (listCostAnalysisRequest) {
-        listCostAnalysisRequest.cancel('Next request has been called.');
-        listCostAnalysisRequest = undefined;
-    }
-    listCostAnalysisRequest = axios.CancelToken.source();
     try {
         costQueryHelper.setFilters(getConvertedFilter(costAnalysisPageState.filters));
         const dateFormat = costAnalysisPageState.granularity === GRANULARITY.MONTHLY ? 'YYYY-MM' : 'YYYY-MM-DD';
         // TODO: Change to clientV2 after the cost analysis API is updated.
-        const { results } = await SpaceConnector.client.costAnalysis.cost.analyze({
+        const { status, response } = await fetchCostAnalyze({
             include_others: !!costAnalysisPageState.chartGroupBy,
             granularity: costAnalysisPageState.granularity,
             group_by: costAnalysisPageState.chartGroupBy ? [costAnalysisPageState.chartGroupBy] : [],
             start: dayjs.utc(costAnalysisPageState.period.start).format(dateFormat),
             end: dayjs.utc(costAnalysisPageState.period.end).format(dateFormat),
             limit: 15,
             ...costQueryHelper.apiQuery,
-        }, { cancelToken: listCostAnalysisRequest.token });
-        listCostAnalysisRequest = undefined;
+        });
         // TODO: Remove conversion process after the cost analysis API is updated.
-        return results.map((d) => ({
-            ...d,
-            cost: d.usd_cost,
-            total_cost: d.total_usd_cost,
-        }));
+        if (status === 'succeed') {
+            return response.results.map((d) => ({
+                ...d,
+                cost: d.usd_cost,
+                total_cost: d.total_usd_cost,
+            }));
+        }
```

Comment:
question: What if status is not 'succeed'?

Reply:
if so, default value will be returned!

```typescript
        }
    } catch (e) {
        ErrorHandler.handleError(e);
    }
    return [];
```